R/LearnerForecast.R

```diff
@@ -1,24 +1,42 @@
 LearnerForecast = R6Class("LearnerForecast", inherit = Learner,
   public = list(
     date_span = NULL,
+    date_frequency = NULL,
     initialize = function(id, param_set = ParamSet$new(), predict_types = "response",
       feature_types = character(), properties = character(), data_formats = "data.table",
       packages = character(), man = NA_character_) {
       super$initialize(id = id, task_type = "forecast", param_set = param_set, feature_types = feature_types,
         predict_types = predict_types, properties = properties, data_formats = data_formats, packages = packages, man = man)
-    },
+      },
 
     train = function(task, row_ids = NULL) {
+      if(is.null(row_ids)){
+        row_ids = task$row_ids
+      }
+      row_ids = sort(row_ids)
       super$train(task, row_ids)
       span = range(task$date(row_ids)[[task$date_col]])
-      self$date_span = list(begin = span[1], end = span[2])
+      self$date_span =
+        list(begin=list(time = span[1], row_id = head(row_ids,1)), end = list(time = span[2], row_id = tail(row_ids,1)))
+      self$date_frequency = time.frequency(task$date(row_ids)[[task$date_col]])
+
+      if(length(self$date_frequency)>1){
+        warning("The timestamps are not equidistant.")
+      }
+    },
```

Comment:
Timestamps don't necessarily need to be equidistant. Check out chapter 2 of Advances in Financial Machine Learning. There he goes over defining the periods by the amount of ticks (trades) that happen in a stock instead of by time. So, for example, you could aggregate every 1000 trades then take the timestamp as the end of that window

Reply:
The purpose of this warning is to inform the user that there may be something wrong. In case of such an aggregation it can just be ignored.